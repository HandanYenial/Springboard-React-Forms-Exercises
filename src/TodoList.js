import React from 'react'
import Todo from './Todo'
import NewTodoForm from './NewTodoForm'
import {useState} from 'react'
import { v4 as uuid } from 'uuid';


function TodoList() {
    const initialTodo = [
        {
        id: uuid(),
        text: "",
        completed: false
      },

        {
        id: uuid(),
        text: "",
        completed: false
        },
    ]

    const [todos, setTodos] = useState(initialTodo);
    const addTodo = (newTodo) =>{
        setTodos(todos =>[...todos, {...newTodo, id: uuid()}])
    }

    const complete = id =>{
        setTodos(todos =>todos.filter( todo => todo.id !== id));
    }
  return (
    <div className= "Todobody">
        <h1 className= "Todo-title"> TodoList </h1>
        {/*Add  the form, so it will display here*/}
        <NewTodoForm addTodo = {addTodo}/>
        {/**Display all todos here */}
        <div>
            {todos.map(({ id, text, completed}) =>
            <Todo 
                key ={id}
                id = {id}
                text = {text}
                completed ={complete}
            />
            )}
        </div>   
    </div>
  );
}

export default TodoList