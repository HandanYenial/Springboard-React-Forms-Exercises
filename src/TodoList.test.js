import React from "react";
import { render ,fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

function addTodo (todoList, text="learn react" , completed="false"){
    const textInput = todoList.screen.getByLabelText("text");
    
    fireEvent.change(textInput, {target: {value: text}});

    const button = todoList.screen.getByText("Add");
    fireEvent.click(button);
}

it("can add a todo" , function(){
    const view = render(<TodoList />);
    addTodo(view);

    expect(view.screen.getByText("learn react")).toBeInTheDocument();
    expect(view.screen.getByText("completed")).toBeInTheDocument();

});

it ("can delete a todo" , function(){
    const view = render(<TodoList/>);
    addTodo(view);

    fireEvent.click(view.screen.getByText("Completed"));

    expect(view.screen.queryByText("learn react")).not.toBeInTheDocument();
})

