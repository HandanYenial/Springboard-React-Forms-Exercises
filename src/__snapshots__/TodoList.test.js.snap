// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todobody"
  >
    <h1
      class="Todo-title"
    >
       TodoList 
    </h1>
    <div>
      <form>
        <label
          for="todo"
        >
           Todo 
        </label>
        <input
          id="todo"
          name="text"
          placeholder="Todo"
          type="text"
          value=""
        />
        <button
          type="submit"
        >
           Add Todo 
        </button>
      </form>
    </div>
    <div>
      <div>
        <h3>
            
        </h3>
        <button>
           Complete 
        </button>
      </div>
      <div>
        <h3>
            
        </h3>
        <button>
           Complete 
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
